src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+import java.util.Set;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport(Storage storage) {
```

Comment:
According to the encapsulation principle, you should not pass the whole Storage object to the getReport method. Instead, consider passing the necessary data directly, such as a Map representing the current state of the storage.